docs/new_services.rst

```diff
@@ -19,3 +23,131 @@ Each service is implemented as a class and should at least implement the followi
 These may block to ensure services start or stop properly, but must *not* block for the full lifetime of the service. If you need to run a blocking process (e.g. run a process via SSH and iterate over its output), this should be done in a background thread. For services that exit after completing a fixed operation (e.g. produce N messages to topic foo), you should also implement ``wait``, which will usually just wait for background worker threads to exit. The ``Service`` base class provides a helper method ``run`` which wraps ``start``, ``wait``, and ``stop`` for tests that need to start a service and wait for it to finish. You can also provide additional helper methods for common test functionality. Normal services might provide a ``bounce`` method.
 
 Most of the code you'll write for a service will just be series of SSH commands and tests of output. You should request the number of nodes you'll need using the ``num_nodes`` or ``node_spec`` parameter to the Service base class's constructor. Then, in your Service's methods you'll have access to ``self.nodes`` to access the nodes allocated to your service. Each node has an associated :class:`~ducktape.cluster.remoteaccount.RemoteAccount` instance which lets you easily perform remote operations such as running commands via SSH or creating files. By default, these operations try to hide output (but provide it to you if you need to extract some subset of it) and *checks status codes for errors* so any operations that fail cause an obvious failure of the entire test.
+
+.. _service-example-ref:
+
+New Service Example
+===================
+
+Let’s walk through an example of writing a simple Zookeeper service.
+
+.. code-block:: python
+
+    class ZookeeperService(Service):
+        PERSISTENT_ROOT = "/mnt"
+        LOG_FILE = os.path.join(PERSISTENT_ROOT, "zk.log")
+        DATA_DIR = os.path.join(PERSISTENT_ROOT, "zookeeper")
+        CONFIG_FILE = os.path.join(PERSISTENT_ROOT, "zookeeper.properties")
+
+        logs = {
+            "zk_log": {
+                "path": LOG_FILE,
+                "collect_default": True},
+            "zk_data": {
+                "path": DATA_DIR,
+                "collect_default": False}
+        }
+
+        def __init__(self, context, num_nodes):
+            super(ZookeeperService, self).__init__(context, num_nodes)
+
+
+Log files will be collected on both successful and failed test runs, while files from the data directory will be collected only on failed test runs. Zookeeper service requests the number of nodes passed to its constructor by passing ``num_nodes`` parameters to the Service base class’s constructor.
```

Comment:
Hopefully they can pick this up from the declarations, but you could add references to the `logs` field and `collect_default` parameters that might make this very clear.